src/libraries/System.Formats.Cbor/src/System/Formats/Cbor/Writer/CborWriter.cs

```diff
@@ -205,11 +205,19 @@ private void EnsureWriteCapacity(int pendingCount)
                 throw new OverflowException();
             }
 
-            if (_buffer is null || _buffer.Length - _offset < pendingCount)
+            int currentCapacity = _buffer is null ? 0 : _buffer.Length;
+            int requiredCapacity = _offset + pendingCount;
+            if (currentCapacity < requiredCapacity)
             {
-                const int BlockSize = 1024;
-                int blocks = checked(_offset + pendingCount + (BlockSize - 1)) / BlockSize;
-                Array.Resize(ref _buffer, BlockSize * blocks);
+                int newCapacity = currentCapacity == 0 ? 1024 : currentCapacity * 2;
+                if ((uint)newCapacity > (uint)Array.MaxLength)
+                    newCapacity = Array.MaxLength;
```

Comment:
This package builds for netstandard2.0. I don't believe `MaxLength` is available there.

Reply:
Ah right, should I replace it with some hardcoded constant like 1GB then?

Reply:
I would replace it with the `Array.MaxValue` constant where it's unavailable; I think that's the correct value for basically all the runtimes we've had anyway isn't it?

Reply:
For both .NET and .NET Framework, the array length limit when the array is a byte[] is 0x7FFFFFC7. It's probably reasonable to do 

```C#
const int MaxArrayLength = 0x7FFFFFC7; // Array.MaxLength
```

Or if we don't feel like burning it in as a const for the latest runtime:


```C#
#if NET
    int maxArrayLength = Array.Length;
#else
    int MaxArrayLength = 0x7FFFFFC7; // Array.MaxLength
#endif
```